src/main/java/com/urbanairship/datacube/idservices/HBaseIdService.java

```diff
@@ -4,221 +4,181 @@
 
 package com.urbanairship.datacube.idservices;
 
-import java.io.IOException;
-import java.nio.ByteBuffer;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Map.Entry;
-
-import org.apache.commons.codec.binary.Base64;
+import com.google.common.base.Optional;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.math.LongMath;
+import com.urbanairship.datacube.IdService;
+import com.urbanairship.datacube.Util;
+import com.urbanairship.datacube.dbharnesses.WithHTable;
+import com.urbanairship.datacube.dbharnesses.WithHTable.ScanRunnable;
 import org.apache.commons.codec.binary.Hex;
 import org.apache.commons.lang.ArrayUtils;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.hbase.HBaseConfiguration;
-import org.apache.hadoop.hbase.client.Delete;
 import org.apache.hadoop.hbase.client.Get;
 import org.apache.hadoop.hbase.client.HTablePool;
 import org.apache.hadoop.hbase.client.Put;
 import org.apache.hadoop.hbase.client.Result;
 import org.apache.hadoop.hbase.client.ResultScanner;
 import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.util.Bytes;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.Multimap;
-import com.google.common.math.LongMath;
-import com.urbanairship.datacube.IdService;
-import com.urbanairship.datacube.Util;
-import com.urbanairship.datacube.dbharnesses.WithHTable;
-import com.urbanairship.datacube.dbharnesses.WithHTable.ScanRunnable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map.Entry;
 
 public class HBaseIdService implements IdService {
     private static final Logger log = LoggerFactory.getLogger(HBaseIdService.class);
-    
+
     public static final byte[] QUALIFIER = ArrayUtils.EMPTY_BYTE_ARRAY;
-    public static final long ALLOC_TIMEOUT_MS = 10000; 
+    public static final int MAX_COMPAT_TRIES = 20;
```

Comment:
oh compatibility. Basically giving us a big ol' window in which both cube versions can live in the same... service? cluster?